nucleus-api/src/main/java/io/github/nucleuspowered/nucleus/api/query/QueryDataCache.java

```diff
@@ -0,0 +1,32 @@
+package io.github.nucleuspowered.nucleus.api.query;
+
+/**
+ * A data cache which stores objects constructed from data retrieved from a query.
+ *
+ * @param <K> The type of the key used to retrieve cached objects.
+ * @param <V> The type of the object being cached.
+ */
+public interface QueryDataCache<K, V> {
```

Comment:
But then, we don't need to provide that system, do we? If another plugin wants to use their own caching system, they should create their own - we're not a utility library.